lnwallet/rpcwallet/rpcwallet.go

```diff
@@ -782,6 +782,28 @@ func (r *RPCKeyRing) MuSig2CombineSig(sessionID input.MuSig2SessionID,
 	return finalSig, resp.HaveAllSignatures, nil
 }
 
+// MuSig2Cleanup removes a session from memory to free up resources.
+func (r *RPCKeyRing) MuSig2Cleanup(sessionID input.MuSig2SessionID) error {
+	req := &signrpc.MuSig2CleanupRequest{
+		SessionId: sessionID[:],
+	}
+
+	ctxt, cancel := context.WithTimeout(context.Background(), r.rpcTimeout)
+	defer cancel()
+
+	_, err := r.signerClient.MuSig2Cleanup(ctxt, req)
+	if err != nil {
+		err = fmt.Errorf("error cleaning up MuSig2 session in remote "+
+			"signer instance: %v", err)
+
+		// Log as critical as we should shut down if there is no signer.
+		log.Criticalf("RPC signer error: %v", err)
```

Comment:
Is the critical error justified here? We could just call with an invalid id and it'd trigger a restart.

Reply:
Ah, good catch! I guess that goes for all the forwarded RPC calls. I think we'll need something like a `shutdownIfUnavailable()` call that only logs the `Criticalf` message if it's a gRPC connection error.

Reply:
Yeah I think I overlooked those when reviewing. The proposed solution sgtm.

Reply:
Added a `considerShutdown` to the error case of all remote signing RPC calls.